search.py

```diff
@@ -56,59 +48,76 @@ def findSimilarVectors(user_tuple):
     choices = ["one", "another", "yet another", "another"]
     for row in results:
         page_contents, similarity_score = row
-        intro = ("Here are the details of " + choices[i] + " relevant opportunity with a similarity score of " + str(similarity_score) + ":\n")
-        output += intro 
-        output += (str(page_contents) + "\n")
+        intro = (
+            "Here are the details of "
+            + choices[i]
+            + " relevant opportunity with a similarity score of "
+            + str(similarity_score)
+            + ":\n"
+        )
+        output += intro
+        output += str(page_contents) + "\n"
         # print("Page Contents: ", page_contents)
         # print("Similarity Score: ", similarity_score)
         i += 1
 
     # Close cursor and connection
     cursor.close()
-    connection.close()
 
     return user_tuple[1] + output
 
 
 def ranker(vector):
     """
     This function performs a similarity search on the embeddings in the database and returns the 25 most similar opportunities
-    
+
     Args:
         user_tuple (tuple): A tuple containing the vectorized version of the fake RFP that was created and the user's query. (vectorized fake rfp, query)
-        
+
     Returns:
         list: A list of the 25 most similar opportunities
     """
     # Generic connection to PostgreSQL
-    connection = psycopg2.connect(
-        host="localhost",
-        port="5432",
-        database="totem",
-        user="postgres"
-    )
 
     # Create a cursor
     cursor = connection.cursor()
 
     # Perform cosine similarity search
     # Returns the page contents of the 25 most similar opportunities
+    # insert_query = """
+    # SELECT page_contents, (embeddings <=> (%s::vector)) AS cosine_distance
+    # FROM totemembeddings
+    # ORDER BY cosine_distance
+    # LIMIT 25;
+    # """
     insert_query = """
-    SELECT page_contents, (embeddings <=> (%s::vector)) AS cosine_distance
-    FROM totemembeddings
-    ORDER BY cosine_distance
-    LIMIT 25;
+    SELECT
+        rowid,
+        distance
+      FROM embeddings
+      WHERE embedding MATCH ?
+      ORDER BY distance
+      LIMIT 25
```

Comment:
Have to first calculate the embedding distance on one table, then use the rowid to get the page_contents from the other table.